File: FranglaisChat/wwwroot/js/LanguageDictionary.js
var langDictionary = (function ()
{
    var dictionary = [];

    dictionary["en-US"] = "English (US)";
    dictionary["en-GB"] = "English (UK)";    
    dictionary["fr-FR"] = "French (France)";
    dictionary["fr-CA"] = "French (Canada)";
    dictionary["es-MX"] = "Spanish (Mexico)";
    dictionary["es-ES"] = "Spanish (Spain)";
    dictionary["de-DE"] = "German";
    dictionary["it-IT"] = "Italian";
    dictionary["ru-RU"] = "Russian";
    dictionary["ja-JP"] = "Japanese";
    //dictionary[""] = "";

    function GetDDLOptions(ddl)
    {
        Object.keys(dictionary).forEach(function (key)
        {
            var option = document.createElement("option");

            option.text = dictionary[key];
            option.value = key;

            ddl.appendChild(option);
        });
    };

    return {
        GetOptions: GetDDLOptions,
        Languages: dictionary
    }
})();

File: FranglaisChat/LanguageDictionary.cs
using System.Collections.Generic;

namespace FranglaisChat
{
    public static class LanguageDictionary
    {
        public static Dictionary<string, string> Languages = new Dictionary<string, string>
        {
            { "en-US", "English (US)" },
            { "en-GB", "English (UK)" },
            { "fr-FR", "French (France)" },
            { "fr-CA", "French (Canada)" },
            { "es-MX", "Spanish (Mexico)" },
            { "es-ES", "Spanish (Spain)" },
            { "de-DE", "German" },
            { "it-IT", "Italian" },
            { "ru-RU", "Russian" },
            { "ja-JP", "Japanese" }
        };
    }
}
